### PR TITLE
Addition: Ingame time GUI element

### PR DESCRIPTION
**Pull request by RayDeeUx**

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GameOverlayRenderEvent) {
+        if (!isEnabled()) return
+
+        config.inGameDateDisplayPosition.renderString(display, posLabel = "In-game Date Display")
+    }
+
+    @SubscribeEvent
+    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt]: Unless im going crazy i dont think a config fix event is needed as these are new features already in the correct spot

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt:42]: to be fair i just cloned one of the display `.kt` files and rolled with it

Comment (CalMWolfs) [reply src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt]: yeah this def aint needed

File: src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java [hunk 2]
```diff
@@ -56,6 +56,26 @@ public static class TextBoxConfig {
     @Expose
     public Position realTimePosition = new Position(10, 10, false, true);
 
+    @Expose
+    @ConfigOption(name = "In-game Date", desc = "Show the in-game date of Skyblock (like in Apec, §ebut with mild delays§7).\n(Though this one includes the Skyblock year!)")
```
Comment (Absterge) [inline src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java]: Both uses of "Skyblock" should be "SkyBlock" for consistency sake.

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java:60]: a simple control+F says yes

however, observe the lowercase `B` the admins used on akinsoft: https://sky.shiiyu.moe/stats/Akinsoft/Not%20Allowed%20To%20Quit%20Skyblock%20Ever%20Again

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java:60]: but fine, i'll revise it

File: src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt (hunk 1 above)
Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt]: removed

File: src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java [hunk 3]
```diff
@@ -56,6 +56,26 @@ public static class TextBoxConfig {
...
+    @Expose
+    @ConfigOption(name = "In-game Date Refresh Rate", desc = "Change the amount of time in seconds you would like to refresh the in-game time display.")
+    @ConfigEditorSlider(
+            minValue = 1,
+            maxValue = 60,
+            minStep = 1
+    )
+    @ConfigAccordionId(id = 2)
+    public int inGameDateDisplayRefreshSeconds = 10;
+
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java]: Please make this into a section in the config (see above this in the file) and also remove the configAccoridianId as its not needed here

File: src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java (hunk 2 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/config/features/GUIConfig.java]: I prefer the wording of the game mode in the Hypixel game selector for everyone  over an troll/Easter egg/bullying/one time joke for one player.

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**